### DISCUSSION TITLE
How do I access GraphQLInputField in the resolver

### DISCUSSION THREAD

**Discussion by swatkatz** [mirumee/ariadne#503] [General]
Hi, I have a directive on the input fields in the schema that subclasses the base GraphQLInputField and adds an additional parameter to the subclass. I have the directive working and I see that the schema present in the GraphQLResolveInfo has the appropriate types.
For e.g.
```
directive @random on INPUT_FIELD_DEFINITION
type Mutation {
      personUpdate(input: PersonUpdateInput!): String
}
input PersonUpdateInput {
      name: String @random
}
```
In python:

```
class RandomInputField(GraphQLInputField):
       def __init__(self, orig_field, random_str):
              super().__init__(...)
              # I want to access this in my resolver execution
              self.random_str = random_str
class Random(SchemaDirectiveVisitor):
def visit_input_field_definition(
        self, field: GraphQLInputField, object_type: Union[GraphQLObjectType, GraphQLInterfaceType]
    ) -> GraphQLInputField:
         return RandomInputField(field, "this random string is added")
mutation = MutationType()
@mutation.field("personUpdate")
def resolve_personUpdate(_, info, input):
       # How do I access the input field here???
       # I want to match the input dictionary with the type of the field, i.e. if the input is {name: "xxx"} and if name was
       # of the type RandomInputField, I want to run some logic
```

**Comment by rafalp** [comment]
Hello!

I've been looking at this in past, when there was a plan for Ariadne to provide `InputType` object allowing developers to custom mappings for input, and I've found no way to achieve this through the `GraphQLInput`/`GraphQLInputField`. There were promising attributes on those types, but I've found those not being used by GraphQL in a useful manner.

**Reply by swatkatz** [reply]
@rafalp : Thanks for the response. I was wondering that the input dictionary does get created somewhere right and if I provide incorrect values in the input, then it does get validated against the specified `InputType`. Is there a way to access that to change the values in the dictionary or provide some extra information?

**Reply by rafalp** [reply]
In theory you could implement custom `ObjectType` class that introspects field arguments types and wraps resolvers in functions that could perform this mechanic for you. But seems like very complex thing to do.

**Reply by swatkatz** [reply]
Gotcha! yeah I got around this by creating some rules around types that would let me pluck them from the typemap on the schema which is present in the info object.
